.github/workflows/publish_images.yaml

```diff
@@ -70,17 +70,29 @@ jobs:
             charmcraft release "${charm_name}" --revision=${charm_rev} --channel=edge --resource="${charm_resource}:${img_rev}"
           }
 
-          for release in $(ls $GITHUB_WORKSPACE/legend/releases/)
-                    
-          do
-            # If there was a new Legend release yesterday, release it to Charmhub
-            if [[ "$release" == $(date +%Y-%m-%d -d "yesterday") ]];          
+          previous_release=""
+          
+          for release in $(ls $GITHUB_WORKSPACE/legend/releases/ | sort -r)
+          
+           do
+            echo "Treating ${release}..."
+            # If the current release was already published, skip.
+            if git show-ref --verify --quiet "refs/heads/release-${release}"
             then
-              echo "Release ${release} is from YESTERDAY and will be uploaded to Charmhub"
-            else
               continue
             fi
-
+            if [ -n "$previous_release" ]
```

Comment:
I don't think we need this if. If it's an empty string, you'll be comparing it with a regular release date anyways, which will be false. This would simplify the code a bit.